### ISSUE TITLE
Instead of the number of the buffer the file path is shown

### ISSUE THREAD

**Issue by rBrda** [mengelbrecht/lightline-bufferline#67]
**OS:** MacOS 10.14.6
**Editor:** Neovim v0.4.4

I have an issue when the following settings are applied in my init.vim:
```
let g:lightline#bufferline#unnamed = '[No Name]'
let g:lightline#bufferline#unicode_symbols = 0
let g:lightline#bufferline#enable_devicons = 1
let g:lightline#bufferline#shorten_path = 1
let g:lightline#bufferline#show_number = 1
let g:lightline#bufferline#smart_path = 0
```

The issue only appears when I turn `lightline#bufferline#smart_path` off.

I also attached a screenshot:
<img width="1018" alt="Screenshot 2020-10-28 at 8 07 58 PM" src="https://user-images.githubusercontent.com/22145465/97484732-6dcf0680-1959-11eb-8f90-a85db85051d3.png">

I was able to do a quick dirty fix in the plugin by modifying the following on line 74 in `autoload/lightline/bufferline.vim`. I basically wrapped the `a:buffer` with `bufnr()`, but I'm not sure if this is the right way to deal with the issue. I'm not experienced with vimscript.

Here is what I did:
```
let l:name = bufnr(a:buffer) . s:number_separator . l:name
```

**Comment by rBrda** [comment]
I tried one more thing. I changed the line on 179 as the following:
```
let [l:name, l:len] = s:get_buffer_name(l:i, bufnr(a:buffers[l:i]), l:buffer_paths[l:i]) 
```
This one is actually the same I mentioned, but it's done when `s:get_buffer_name()` gets called.

**Comment by mengelbrecht** [comment]
Thanks for the report. Your investigation helped me find the cause quickly 👍. I pushed a fix in 3933dcd.
